Assets/Scripts/Health.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Health : MonoBehaviour
+{
+    [SerializeField]
+    public float maxHealth;
+    public float currentHealth;
+
+    [SerializeField]
+    private UnityEvent onDeathActivated;
+    private bool triggeredDeath = false;
+    
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = maxHealth;
+    }
+    public void LoseHealth(float amount)
+    {
+        Debug.Log("Lost health:" + gameObject.name);
+        //Losing Health
+        currentHealth -= amount;
+
```

Comment:
After reducing health
If the slider has a range of 0, 1: `healthBar.value = currentHealth/maxHealth`
If the slider has a range of 0, maxHealth: `healthBar.value = currentHealth`